tensorflow/c/kernels.cc

```diff
@@ -272,3 +272,31 @@ TF_Tensor* TF_AllocateOutput(TF_OpKernelContext* context, int index,
   }
   return tf_tensor;
 }
+
+TF_Tensor* TF_AllocateTemp(TF_OpKernelContext* context, TF_DataType dtype, 
+                           int64_t* dims, int num_dims, 
+                           TF_AllocatorAttributes attributes,
+                           TF_Status* status) {
+  auto* cc_ctx = reinterpret_cast<::tensorflow::OpKernelContext*>(context);
+  TF_SetStatus(status, TF_OK, ""); 
+  tensorflow::TensorShape shape;
+  for (int i = 0; i < num_dims; ++i) {
+    shape.AddDim(dims[i]); 
+  }
+  tensorflow::AllocatorAttributes allocator_attr; 
+  if (attributes.on_host) { 
+    allocator_attr.set_on_host(true); 
+  }
+  tensorflow::Status s;
+  tensorflow::Tensor tensor_temp;  
+  TF_Tensor* tf_tensor_temp; 
+  s = cc_ctx->allocate_temp(static_cast<tensorflow::DataType>(dtype), shape, 
+  		&tensor_temp, allocator_attr);
+  if (s.ok()) { 
+    tf_tensor_temp = TF_TensorFromTensor(tensor_temp, &s); 
+  }
+  if (s.ok()) { 
+    ::tensorflow::Set_TF_Status_from_Status(status, s); 
+    return tf_tensor_temp; 
```

Comment:
This looks like the old implementation you had in previous PR. You updated it it since then: https://github.com/tensorflow/tensorflow/pull/40854/files
So, it should be updated here too.